python/Scripts/creatematerial.py

```diff
@@ -0,0 +1,235 @@
+            normal_map.setConnectedNode('in', image_node)
+            conn_node = normal_map
+
+        output_node = node_graph.addOutput(texture_name + '_output', plug_type)
+        output_node.setConnectedNode(conn_node)
+        mtl_input.setConnectedOutput(output_node)
+        mtl_input.setType(plug_type)
+
+        if texture_file.isUDIM():
+            udim_numbers.update(set(texture_file.getUdimNumbers()))
+
+    # Create udim set
+    if udim_numbers:
+        geom_info_name = doc.createValidChildName('GI_' + mtlx_filename)
+        geom_info = doc.addGeomInfo(geom_info_name)
+        geom_info.setGeomPropValue('udimset', list(udim_numbers), "stringarray")
+
+    # Write mtlx file
+    if not mtlx_file.getParentPath().exists():
+        mtlx_file.getParentPath().createDirectory()
+
+    MaterialX.writeToXmlFile(doc, mtlx_file.asString())
+    logger.info("MaterialX file created `{}`".format(mtlx_file.asString()))
+
+
+def main():
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument('-o', '--outputFilename', dest='outputFilename', help='Filename of the output materialX document (default material.mtlx).')
+    parser.add_argument('-c', '--colorSpace', dest='colorSpace', help='Colorsapce to set (default to `srgb_texture`).')
+    parser.add_argument('-a', '--absolutePaths', dest='absolutePaths', action="store_true",help='Make the texture paths absolute inside the materialX file.')
+    parser.add_argument('-t', '--tileimage', dest='tileimage', action="store_true",help='Use tileimage node instead of image node.')
+    parser.add_argument('-v', '--verbose', dest='verbose', action="store_true",help='Turn on verbose mode to create loggings.')
+    parser.add_argument(dest='inputDirectory', nargs='?', help='Directory that contain textures (default to current working directory).')
+    # TODO : Flag for SG names to be created in mtlx file. default, djed SG pattern or first match.
+    # TODO : Flag to seperate each SG for mtlx with the name of each shading group, default combined in one mtlx file name by output file name.
+    # TODO : Flag for forcing texture extension if there are multiple extensions in directory.
+
+    options = parser.parse_args()
+
+    texture_path = MaterialX.FilePath.getCurrentPath()
+
+    if options.inputDirectory:
+        texture_path = MaterialX.FilePath(options.inputDirectory)
+
+        if not texture_path.isDirectory():
+            logger.error("The texture directory does not exist `{}`".format(texture_path))
+            return
+
+    default_doc_name = MaterialX.FilePath('standard_surface.mtlx')
+    mtlx_file = texture_path / default_doc_name
+    if options.outputFilename:
+        filepath = MaterialX.FilePath(options.outputFilename)
+
+        if filepath.isAbsolute():
+            mtlx_file = filepath
+        else:
+            mtlx_file = texture_path / filepath
+
+    # Colorspace
+    colorspace = 'srgb_texture'
```

Comment:
Curious why is default colorspace sRGB ? Perhaps there should be no mapping by default ?

Reply:
Do you mean no colorspace is better than sRGB?